test-lib.sh

```diff
@@ -40,37 +62,70 @@ UNSTABLE_TESTS="${UNSTABLE_TESTS:-""}"
 function ct_cleanup() {
   ct_show_resources
   ct_show_results
+  ct_clean_app_images
 
   if [[ -z ${CID_FILE_DIR:-} ]]; then
     echo "The \$CID_FILE_DIR is not set. Container cleaning is to be skipped."
     exit "${TESTSUITE_RESULT:-0}"
   fi;
 
-  echo "Examining CID files in $CID_FILE_DIR"
+  echo "Examining CID files in \$CID_FILE_DIR=$CID_FILE_DIR"
   for cid_file in "$CID_FILE_DIR"/* ; do
     [ -f "$cid_file" ] || continue
     local container
     container=$(cat "$cid_file")
+    rm "$cid_file"
 
     echo "Stopping and removing container $container..."
+    ct_container_exists "$container" || continue
     docker stop "$container"
 
     # Container has not been removed by `docker stop` and still exists
-    if [ "$( docker ps -a -f "id=$container" | wc -l )" -eq 2 ]; then
+    if ct_container_exists "$container"; then
```

Comment:
This condition is not needed. You are already checking it a couple of rows above `ct_container_exists "$container" || continue`